typings/index.d.ts

```diff
@@ -3199,6 +3223,34 @@ export type CacheWithLimitsOptions = {
     : never;
 };
 
+export interface CategoryCreateChannelOptions {
+  permissionOverwrites?: OverwriteResolvable[] | Collection<Snowflake, OverwriteResolvable>;
+  topic?: string;
+  type?: Exclude<
```

Comment:
maybe out of acope for this PR, but i think this should have a better type

```ta
type ExcludeEnum<Enum, Keys extends keyof Enum> = Exclude<keyof typeof Enum | Enum, Keys | Enum[Keys]>
```

because as-is this type is unnessercerily big

Reply:
I'm not experienced with types on this level so if you could provide an example of how this would look like I could add it, and since the PR already touches this type I don't see why not

Reply:
```ts
ExcludeEnum<ChannelType,
  'DM'
>
```
instead of
```ts
Exclude<keyof typeof ChannelType | ChannelType,
  'DM',
  ChannelType.DM
>
```

Reply:
Will try this and commit if it goes well

Reply:
the type might need a bit of fine tuning but yea

Reply:
Decided I'll add this in a different PR after this one is merged since I also wanna use this ExcludeEnum type in other places, so I'll mark this as resolved for now